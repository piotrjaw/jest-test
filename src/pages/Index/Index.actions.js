import * as TYPES from './Index.types';
import { del, get, put, post } from '../../helpers/request';

const createTodoPending = () => ({
  type: TYPES.CREATE_TODO_PENDING
});

const createTodoSuccess = () => ({
  type: TYPES.CREATE_TODO_SUCCESS
});

const deleteTodoPending = () => ({
  type: TYPES.DELETE_TODO_PENDING
});

const deleteTodoSuccess = () => ({
  type: TYPES.DELETE_TODO_SUCCESS
});

const getTodosPending = () => ({
  type: TYPES.GET_TODOS_PENDING
});

export const getTodosSuccess = (data) => ({
  type: TYPES.GET_TODOS_SUCCESS,
  data
});

const updateTodoPending = () => ({
  type: TYPES.UPDATE_TODO_PENDING
});

const updateTodoSuccess = () => ({
  type: TYPES.UPDATE_TODO_SUCCESS
});

export const createTodo = (createObj) => (dispatch) => {
  if (!createObj.text) return Promise.reject();
  dispatch(createTodoPending());
  return post('/todos/', createObj)
    .then((res) => dispatch(createTodoSuccess()));
}

export const deleteTodo = (deleteObj) => (dispatch) => {
  dispatch(deleteTodoPending());
  return del('/todos/', deleteObj)
    .then((res) => dispatch(deleteTodoSuccess()));
};

export const getTodos = () => (dispatch) => {
  dispatch(getTodosPending());
  return get('/todos/')
    .then(({ data }) => dispatch(getTodosSuccess(data)));
};

export const updateTodo = (updateObj) => (dispatch) => {
  dispatch(updateTodoPending());
  return put('/todos/', updateObj)
    .then((res) => dispatch(updateTodoSuccess()));
};
