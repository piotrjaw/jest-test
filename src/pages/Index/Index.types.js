export const CREATE_TODO_PENDING = 'Index/CREATE_TODO_PENDING';
export const CREATE_TODO_SUCCESS = 'Index/CREATE_TODO_SUCCESS';
export const DELETE_TODO_PENDING = 'Index/DELETE_TODO_PENDING';
export const DELETE_TODO_SUCCESS = 'Index/DELETE_TODO_SUCCESS';
export const GET_TODOS_PENDING = 'Index/GET_TODOS_PENDING';
export const GET_TODOS_SUCCESS = 'Index/GET_TODOS_SUCCESS';
export const UPDATE_TODO_PENDING = 'Index/UPDATE_TODO_PENDING';
export const UPDATE_TODO_SUCCESS = 'Index/UPDATE_TODO_SUCCESS';
