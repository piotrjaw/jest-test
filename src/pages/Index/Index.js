import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as IndexActions from './Index.actions';
import List from '../../components/List';

class IndexComponent extends PureComponent {
  render() {
    return (
      <div>Index</div>
    );
  }
}

const mapStateToProps = ({ index }) => ({

});

const mapDispatchToProps = (dispatch) => ({
  actions: bindActionCreators(IndexActions, dispatch)
});

const Index = connect(
  mapStateToProps,
  mapDispatchToProps
)(IndexComponent);

export default Index;
