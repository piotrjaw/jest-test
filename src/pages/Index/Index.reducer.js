import * as TYPES from './Index.types';

const initialState = {};

export default function (state = initialState, action) {
  switch (action.type) {
    default:
      return state;
  }
};
