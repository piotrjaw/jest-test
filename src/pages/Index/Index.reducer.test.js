import reducer from './Index.reducer';
import * as TYPES from './Index.types';

let initialState;
let result;

beforeEach(() => {
  initialState = {};
});

afterEach(() => {
  expect(result).toMatchSnapshot();
  result = null;
});

describe('Index reducer', () => {
  it('should return the initial state', () => {
    result = reducer(undefined, {});
    expect(result).toEqual({
      todos: []
    });
    expect(result).toMatchSnapshot();

    result = reducer(initialState, {});
    expect(result).toEqual({
      todos: []
    });
  });

  it('should handle CREATE_TODO', () => {
    result = reducer(initialState, {
      type: TYPES.CREATE_TODO
    });
    expect(result).toEqual({
      todos: []
    });
    expect(result).toMatchSnapshot();

    result = reducer({
      todos: [{ id: 1, text: 'test1' }]
    }, {
      type: TYPES.CREATE_TODO
    });
    expect(result).toEqual({
      todos: [{ id: 1, text: 'test1' }]
    });
  });

  it('should handle GET_TODOS_SUCCESS', () => {
    result = reducer(initialState, {
      type: TYPES.GET_TODOS_SUCCESS,
      data: [
        { id: 1, text: 'test1' },
        { id: 2, text: 'test2' }
      ]
    });
    expect(result).toEqual({
      todos: [
        { id: 1, text: 'test1' },
        { id: 2, text: 'test2' }
      ]
    });
  });
});
