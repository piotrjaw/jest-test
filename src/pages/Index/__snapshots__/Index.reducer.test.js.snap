// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index reducer should handle CREATE_TODO 1`] = `
Object {
  "todos": Array [],
}
`;

exports[`Index reducer should handle CREATE_TODO 2`] = `
Object {
  "todos": Array [
    Object {
      "id": 1,
      "text": "test1",
    },
  ],
}
`;

exports[`Index reducer should handle GET_TODOS_SUCCESS 1`] = `
Object {
  "todos": Array [
    Object {
      "id": 1,
      "text": "test1",
    },
    Object {
      "id": 2,
      "text": "test2",
    },
  ],
}
`;

exports[`Index reducer should return the initial state 1`] = `
Object {
  "todos": Array [],
}
`;

exports[`Index reducer should return the initial state 2`] = `
Object {
  "todos": Array [],
}
`;
